src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -75,18 +99,30 @@ private struct EHEnum
 #pragma warning disable IDE0060
         // This is a fail-fast function used by the runtime as a last resort that will terminate the process with
         // as little effort as possible. No guarantee is made about the semantics of this fail-fast.
-        internal static void FallbackFailFast(RhFailFastReason reason, object unhandledException)
+#if !NATIVEAOT
```

Comment:
Since this appears to be used actively in the CoreCLR case, we should describe the semantics here better, or at least describe the semantics we are relying on.

Reply:
With this change, I wanted to make it easier to review the changes in this file by not refactoring / renaming stuff / moving the file. I'd like to tackle it in a next step.